### ISSUE TITLE
Wrong warning: "Reported kernel version is older than systemd's required baseline kernel version 4.15" with a 5.17 Kernel

### ISSUE THREAD

**Issue by ms178** [systemd/systemd#23059]
**systemd version the issue has been seen with**
 > systemd 251.rc1.r275.gbc85f8b51d-1-arch running in system mode (+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUT>

**Used distribution**
 > EndeavourOS 22.1 (Apollo)

**Linux kernel version used** (`uname -a`)
 > `5.17.2-Xanmod`

**CPU architecture issue was seen on**
 > Intel Xeon 2696V3 (Haswell-EP)

**Expected behaviour you didn't see**
 > Not seeing a warning about the Kernel version as it is brand new

**Unexpected behaviour you saw**
 > Warning about an "old" Kernel version and potentially reduced functionality due to this mis-identification

**Steps to reproduce the problem**
 > Compile the systemd-git package from AUR, I use a slightly modified PKGBUILD (attached). This is a regression with systemd 251 as I don't see this issue with my custom 5.17-Kernel with the default systemd 250 that EndeavourOS ships with.
 
[PKGBUILD.txt](https://github.com/systemd/systemd/files/8473691/PKGBUILD.txt)

**Additional program output to the terminal or log subsystem illustrating the issue**

```
[  +0,005507] systemd[1]: systemd 251.rc1.r275.gbc85f8b51d-1-arch running in system mode (+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUT>
[  +0,012444] systemd[1]: Detected architecture x86-64.
[  +0,000175] systemd[1]: Warning! Reported kernel version  is older than systemd's required baseline kernel version 4.15. Your mileage may vary.
```

…
```

**Comment by poettering** [comment]
Ouch, I used assert() instead of assert_se() around the uname()

That said, please talk to your distro. They should not set NDEBUG. That's just stupid pseudo-optimization

**Comment by yuwata** [comment]
Anyway, thank you for the report. Fix is waiting in #23060.
